src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java

```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
   public static class PhrasalRule {
     List<String> sourceWords;
     List<String> targetWords;
+    double score = 0.0;
+
+    public PhrasalRule(String sourcePhrase, String targetPhrase, double score) {
+      if (sourcePhrase != null && targetPhrase != null) {
+        sourceWords = Arrays.asList(sourcePhrase.split("\\s+"));
+        targetWords = Arrays.asList(targetPhrase.split("\\s+"));
+      } else {
+        sourceWords = Collections.emptyList();
+        targetWords = Collections.emptyList();
+      }
+      this.score = score;
+    }
 
     public String getTarget() {
       return String.join(" ", targetWords);
     }
+
+    public double getScore() {
+      return score;
+    }
+  }
+
+  public class PhrasalRuleComparer implements Comparator<PhrasalRule> {
+    @Override
+    public int compare(PhrasalRule x, PhrasalRule y) {
+      if (x != null && y != null) {
+        return Double.compare(y.score, x.score);
```

Comment:
I think it's the other way around since x and y are switched in the compare call.